### PR TITLE
Added 64 bit version of BSI library.  Add NewBSIRetainSet() method.

### PR DESCRIPTION
**Pull request by guymolinari**

### CODE REVIEW

**Comment by guymolinari** [comment]
@lemire any chance of a review/merge in the next few days?  I need to do a product release and trying not to include any "one off" repos.

**Comment by lemire** [comment]
Got it.

File: BitSliceIndexing/bsi.go [hunk 1]
```diff
@@ -474,3 +474,27 @@ func (b *BSI) ClearValues(foundSet *roaring.Bitmap) {
 	}
 	wg.Wait()
 }
+
+// NewBSIRetainSet
+func (b *BSI) NewBSIRetainSet(foundSet *roaring.Bitmap) *BSI {
+
+        newBSI := NewBSI(b.MaxValue, b.MinValue)
```
Comment (lemire) [inline BitSliceIndexing/bsi.go:481]: Can you run "go fmt" on this code? It is a small thing, and it is optional,  but it does save me the trouble of doing it later.

File: roaring64/bsi64.go [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+
+	var n int = parallelism
+	if n == 0 {
+		n = runtime.NumCPU()
+	}
+
+	resultsChan := make(chan *Bitmap, n)
+
+	card := foundSet.GetCardinality()
+	x := uint64(math.Floor(float64(card) / float64(n)))
```
Comment (lemire) [inline roaring64/bsi64.go:114]: I know that you have used this construction before, but this seems unnecessarily complicated. Given that card and n are non-negative, then you can just do x := card / n, minus some casting. The integer division is already defined as a floor.

File: roaring64/bsi64.go [hunk 3]
```diff
@@ -0,0 +1,498 @@
...
+
+	remainder := card - (x * uint64(n))
```
Comment (lemire) [inline roaring64/bsi64.go:116]: Again, this is just "card  % n".

**Comment by lemire** [comment]
Do not mind the failing travis tests. We are retiring travis.

**Comment by guymolinari** [comment]
Thanks @lemire, changes pushed.

**Comment by lemire** [comment]
Merging.

**Pull request merged**